Kickstarter-iOS/Views/Controllers/MessageBannerViewController.swift

```diff
@@ -0,0 +1,125 @@
+import Foundation
+import Library
+import Prelude
+
+final class MessageBannerViewController: UIViewController {
+  @IBOutlet fileprivate weak var backgroundView: UIView!
+  @IBOutlet fileprivate weak var backgroundViewBottomConstraint: NSLayoutConstraint!
+  @IBOutlet fileprivate weak var iconImageView: UIImageView!
+  @IBOutlet fileprivate weak var messageLabel: UILabel!
+
+  private var bottomMarginConstraintConstant: CGFloat = -Styles.grid(1)
+
+  private let viewModel: MessageBannerViewModelType = MessageBannerViewModel()
+
+  struct AnimationConstants {
+    static let hideDuration: TimeInterval = 0.25
+    static let showDuration: TimeInterval = 0.3
+  }
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    _ = backgroundView
+      |> roundedStyle(cornerRadius: 4)
+
+    _ = messageLabel
+      |> UILabel.lens.font .~ .ksr_subhead()
+  }
+
+  override func bindViewModel() {
+    super.bindViewModel()
+
+    self.iconImageView.rac.hidden = self.viewModel.outputs.iconIsHidden
+    self.backgroundView.rac.backgroundColor = self.viewModel.outputs.bannerBackgroundColor
+    self.messageLabel.rac.text = self.viewModel.outputs.bannerMessage
+    self.messageLabel.rac.textColor = self.viewModel.outputs.messageTextColor
+
+    self.viewModel.outputs.messageBannerViewIsHidden
+      .observeForUI()
+      .observeValues { [weak self] isHidden in
+        self?.showViewAndAnimate(isHidden)
+    }
+
+    self.viewModel.outputs.iconImage
+      .observeForUI()
+      .observeValues { [weak self] image in
+        guard let `self` = self else { return }
+        _ = self.iconImageView
+          |> UIImageView.lens.image .~ image
+    }
+
+    self.viewModel.outputs.messageTextAlignment
+      .observeForUI()
+      .observeValues { [weak self] textAlignment in
+        guard let `self` = self else { return }
+
+        _ = self.messageLabel
+         |> UILabel.lens.textAlignment .~ textAlignment
+    }
+  }
+
+  func setBannerType(type: MessageBannerType, message: String) {
+    self.viewModel.inputs.setBannerType(type: type)
+    self.viewModel.inputs.setBannerMessage(message: message)
+  }
+
+  func showBannerView() {
+    self.viewModel.inputs.showBannerView(shouldShow: true)
+  }
+
+  private func showViewAndAnimate(_ isHidden: Bool) {
+    let duration = isHidden ? AnimationConstants.hideDuration : AnimationConstants.showDuration
+
+    if !isHidden {
```

Comment:
No because when `isHidden` is true, we don't want to hide it before the animation has completed. Only in the scenario where `isHidden` is false, do we want the view's visibility to be updated before the animation starts.